--- a/sge.in	2007-03-14 11:38:26.000000000 +0700
+++ b/sge.in	2007-04-02 17:02:53.000000000 +0700
@@ -30,7 +30,7 @@
 
 my ($qsub, $qstat, $qdel, $qselect, $qhost, $qconf, $qacct,
     $mpirun, $sun_mprun, $mpi_pe, $cat, $supported_job_types,
-    $SGE_ROOT, $SGE_CELL, $SGE_MODE, $SGE_RELEASE);
+    $SGE_ROOT, $SGE_CELL, $SGE_MODE, $SGE_RELEASE, $SGE_QMASTER_PORT);
 
 BEGIN
 {
@@ -57,6 +57,7 @@
     $SGE_CELL    = '@SGE_CELL@';
     $SGE_MODE    = '@SGE_MODE@';
     $SGE_RELEASE = '@SGE_RELEASE@';
+    $SGE_QMASTER_PORT = '536';
 }
 
 
@@ -557,6 +558,7 @@
     #       emptied in GT3.
     $ENV{"SGE_ROOT"} = $SGE_ROOT;
     $ENV{"SGE_CELL"} = $SGE_CELL;
+    $ENV{"SGE_QMASTER_PORT"} = $SGE_QMASTER_PORT;
     chomp($job_id = `$qsub $sge_job_script_name`);
 
     if($? == 0)
@@ -608,6 +610,7 @@
     # [dwm]  Replacement state checking code.
     $ENV{"SGE_ROOT"} = $SGE_ROOT;
     $ENV{"SGE_CELL"} = $SGE_CELL;
+    $ENV{"SGE_QMASTER_PORT"} = $SGE_QMASTER_PORT;
     
     # Get first line matching job id
     $_ = (grep(/^\s+$job_id\s/, `$qstat 2> /dev/null`))[0];
